scripts/test.sh

```diff
@@ -18,41 +18,111 @@
 # Build the quickstart. If we're running on the main repo (not a fork), we
 # also run the tests along with the decoded GoogleService-Info.plist files.
 
-set -eo pipefail
-
-EXIT_STATUS=0
+set -euo pipefail
 
 # Set have_secrets to true or false.
-. ./scripts/check_secrets.sh
+have_secrets=false
+source scripts/check_secrets.sh
+
+# Get Xcode version
+system=$(uname -s)
+case "$system" in
+  Darwin)
+    xcode_version=$(xcodebuild -version | head -n 1)
+    xcode_version="${xcode_version/Xcode /}"
+    xcode_major="${xcode_version/.*/}"
+    ;;
+  *)
+    xcode_major="0"
+    ;;
+esac
+
+# Check Xcode version when testing watchOS
+if [[ "$TEST" == true && \
+      "$OS" == watchOS && \
+      "$xcode_major" -lt 13 && \
+      "$xcode_version" != "12.5.1" && \
+      "$xcode_version" != "12.5" ]]; then
+    echo "Xcode version does not yet supporting testing on watchOS"
+    exit 1
+fi
+
+# Initialize flags
+flags=()
+
+# Set project / workspace
+if [[ "$SPM" == true ]];then
+    flags+=( -project "${DIR}/${SAMPLE}Example.xcodeproj" )
+else
+    if [[ "$LEGACY" == true ]]; then
+        WORKSPACE="${SAMPLE}/Legacy${SAMPLE}Quickstart/${SAMPLE}Example.xcworkspace"
+    else
+        WORKSPACE="${SAMPLE}/${SAMPLE}Example.xcworkspace"
+    fi
+    flags+=( -workspace "$WORKSPACE" )
+fi
+
+# Set scheme
+if [[ "$SPM" == true ]];then
+    SCHEME="${SAMPLE}Example (${OS})"
+else
+    SCHEME="${SAMPLE}Example${SWIFT_SUFFIX}"
+fi
+flags+=( -scheme "$SCHEME" )
+
+# Set destination
+if [[ "$OS" == iOS ]]; then
+    DESTINATION="platform=iOS Simulator,name=${DEVICE}"
+    flags+=( -destination "$DESTINATION" )
+elif [[ "$OS" == tvOS ]]; then
+    DESTINATION="platform=tvOS Simulator,name=${DEVICE}"
+    flags+=( -destination "$DESTINATION" )
+elif [[ "$OS" == macOS || "$OS" == catalyst ]]; then
+    DESTINATION="platform=macos"
+    flags+=( -destination "$DESTINATION" )
+elif [[ "$OS" == watchOS ]]; then
+    DESTINATION="platform=watchOS Simulator,name=${DEVICE}"
+else
```

Comment:
`DESTINATION` is not used when targeting watchOS: previously there was trouble getting `xcodebuild` to use the right simulator / device, and without a `-destination` flag it works fine for the moment.